packages/frontend/src/view/components/safe/utils.ts

```diff
@@ -113,3 +114,114 @@ export function toUTC(dateString: string): string {
   const date = new Date(dateString)
   return date.toUTCString()
 }
+
+export function getDecodedProperties(tx: SafeMultisigTransaction) {
+  const parsingResult = SafeTransactionDecodedData.safeParse(tx.dataDecoded)
+
+  if (parsingResult.success && tx.dataDecoded) {
+    // SafeApi kit typings are wrong, string type applies only for transport layer
+    // string is later parsed into json object
+    const decodedCall = decodeCall(
+      tx.dataDecoded as unknown as NonNullable<SafeTransactionDecodedData>,
+    )
+
+    return {
+      method: decodedCall.method,
+      signature: decodedCall.signature,
+      callWithParams: decodedCall.functionCall,
+      params: decodedCall.params,
+    }
+  }
+
+  return null
+}
+
+export type MultisigOwnershipHistory = ReturnType<
+  typeof deriveMultisigOwnership
+>
+
+export function deriveMultisigOwnership(
+  allTransactions: SafeMultisigTransaction[],
+  latestOwners: number,
+) {
+  const ownerChangingTransactions = allTransactions
+    .map((tx) => ({ raw: tx, decoded: getDecodedProperties(tx) }))
+    .filter(
```

Comment:
Are we sure that all the transactions here have been executed?

Reply:
That's exactly the purpose of line ~153, where we filter only executed transactions